pallets/rewards/src/lib.rs

```diff
@@ -154,7 +162,7 @@ pub mod pallet {
 		Blake2_128Concat,
 		T::AccountId,
 		Blake2_128Concat,
-		T::CurrencyId,
+		(T::DomainId, T::CurrencyId),
```

Comment:
Only "downside" is that when querying storage & iterating over the currencies, clients will need to know that this value is a composed one, since is most likely an enum, it should be easy.

Reply:
Yes, I do not know how Substrate encodes/decodes a tuple of values, but I think should be easy to identify both things.

Reply:
We could also use a `StorageNMap` here and avoid that tuple, but not sure if it's worth it.

Reply:
No, I think is fine with the tuple.